src/state/swap/hooks.ts

```diff
@@ -346,6 +346,11 @@ export const useDefaultsFromURLSearch = ():
 
   const { currencies } = useDerivedSwapInfo()
 
+  const currenciesRef = useRef(currencies)
+  useEffect(() => {
```

Comment:
`currenciesRef.current = currencies` is enough, no need useEffect here

Reply:
make sense